src/video/renderers/opengl3/gl3_renderer.c

```diff
@@ -238,28 +239,34 @@ static void capture_screenshot(void *userdata) {
     omf_free(buffer);
 }
 
+#define ASPECT_X (4.0f / 3.0f)
+#define ASPECT_Y (3.0f / 4.0f)
+
 /**
  * Set the viewport, and do screen-shakes here.
  */
 static inline void set_screen_viewport(const gl3_context *ctx) {
-    float move_ratio = ctx->screen_w / NATIVE_W;
     float native_aspect = ctx->screen_w / ctx->screen_h;
     int vp_x = 0;
     int vp_y = 0;
     int vp_w = ctx->viewport_w;
     int vp_h = ctx->viewport_h;
 
+    // aspect == 0 means 4:3
+    // aspect == 1 means stretch to window
     if(ctx->aspect == 0) {
-        // Set 4:3 aspect ratio, as requested
         if(native_aspect >= 1.0f) {
-            vp_w = ctx->viewport_h * 4 / 3;
-            vp_x = (ctx->viewport_w - vp_w) / 2;
+            vp_w = min2(ctx->screen_w, ctx->viewport_h * ASPECT_X);
+            vp_h = vp_w * ASPECT_Y;
         } else {
-            vp_h = ctx->viewport_w * 3 / 4;
-            vp_y = (vp_h - ctx->viewport_h) / 2;
+            vp_h = min2(ctx->screen_h, ctx->viewport_w * ASPECT_Y);
+            vp_w = vp_h * ASPECT_X;
         }
+        vp_x = (ctx->viewport_w - vp_w) / 2;
+        vp_y = (ctx->viewport_h - vp_h) / 2;
     }
 
+    float move_ratio = ctx->screen_w / NATIVE_W;
```

Comment:
```suggestion
    float move_ratio = vp_w / NATIVE_W;
```
to scale screenshake with game viewport size instead of window width

Reply:
I don't know, actually. I left it as it was, as it seemed subjectively better.

Reply:
I'm willing to entertain other opinions ;)

Reply:
I ended up wiring it to the aspect corrected image. Probably closer to the original.